server/journeys/screen/screen/tests.test.ts

```diff
@@ -0,0 +1,156 @@
+    const html = createTestHtmlElement(result.text)
+
+    expect(getByRole(html, 'heading', { name: /Screen a CSIP referral/ })).toBeVisible()
+    expect(getByRole(html, 'heading', { name: /Describe the reasons for this decision/ })).toBeVisible()
+
+    const radios = getAllByRole(html, 'radio')
+    radios.forEach(radio => {
+      expect(radio).not.toBeChecked()
+    })
+
+    expect(getByRole(html, 'button', { name: /continue/i })).toBeVisible()
+
+    expect(queryByAttribute('class', html, 'govuk-breadcrumbs')).toBeNull()
+    expect(queryByRole(html, 'link', { name: 'Back' })).toBeNull()
+  })
+
+  it('should prepopulate values', async () => {
+    const data = {
+      journeyData: {
+        instanceUnixEpoch: 0,
+        saferCustodyScreening: {
+          outcomeType: { code: 'WFA', description: 'No further action' } as ReferenceData,
+          reasonForDecision: 'idklol',
+        },
+      },
+    }
+
+    const result = await request(app(data)).get(`/${uuid}/screen/screen`).expect(200).expect('Content-Type', /html/)
+
+    const html = createTestHtmlElement(result.text)
+
+    const radios = getAllByRole(html, 'radio')
+
+    expect(radios[0]).toBeChecked()
+    expect(radios[1]).not.toBeChecked()
+
+    expect(html.getElementsByTagName('textarea')[0]!.value).toBe('idklol')
+
+    expect(getByRole(html, 'button', { name: /continue/i })).toBeVisible()
+  })
+
+  it('should redirect on posting empty data', async () => {
+    await request(app()).post(`/${uuid}/screen/screen`).send({}).expect('Location', `/`)
+
+    expect(reqCaptured.validationErrors()).toEqual({
+      saferCustodyOutcome: ['Select the outcome of Safer Custody screening'],
+      reasonForDecision: ['Enter a description of the reasons for this decision'],
+    })
+  })
+
+  it('should redirect on posting invalid data', async () => {
+    await request(app())
+      .post(`/${uuid}/screen/screen`)
+      .send({
+        saferCustodyOutcome: 'INVALID',
+        reasonForDecision: 'Posting a value for outcomeType that should not be accepted',
+      })
+      .expect('Location', `/`)
+
+    expect(reqCaptured.validationErrors()).toEqual({
+      saferCustodyOutcome: ['Select the outcome of Safer Custody screening'],
+    })
+  })
+
+  it('should redirect on posting over 4000 characters in reason', async () => {
+    await request(app())
+      .post(`/${uuid}/screen/screen`)
+      .send({
+        reasonForDecision: 'o'.repeat(4001),
+        saferCustodyOutcome: 'WFA',
+      })
+      .expect('Location', `/`)
+
+    expect(reqCaptured.validationErrors()).toEqual({
+      reasonForDecision: ['Description must be 4,000 characters or less'],
+    })
+  })
+
+  it('should return a 200 on posting valid data and redirect to check answers', async () => {
+    await request(app())
+      .post(`/${uuid}/screen/screen`)
+      .send({ saferCustodyOutcome: 'WFA', reasonForDecision: 'no action needed' })
+      .expect(302)
+      .expect('Location', 'check-answers')
+  })
```

Comment:
Would be great to get a test of the validation messages appearing on the screen in the different combinations possible - have a look at `server/journeys/referral/involvement/tests.test.ts` (the last test) for an example if you like.

Reply:
On subtle test that's missing is one that covers one input value being valid and one not. Both values should be retained and replayed back to the user. In this case, I'd selected "No further action" and not entered anything for reasons and it didn't retain my outcome selection. It's due to the `res.locals.formResponses?.['saferCustodyOutcome'] || req.journeyData. saferCustodyScreening?.outcomeType?.code` PR comment above and validation tests would have caught it.

To be fair, I think we have other pages that have similar missing tests and possible bugs

![image](https://github.com/user-attachments/assets/073955d3-73d1-40ce-973e-1acfcf602386)

Reply:
Good catch. This is now covered by the cypress e2e tests